Eatery Blue/UI/Carousel/CarouselView.swift

```diff
@@ -140,64 +139,87 @@ class CarouselView: UIView {
 
         navigationController?.pushViewController(viewController, animated: true)
     }
-}
 
-extension CarouselView: UICollectionViewDataSource {
-    
-    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
-        if indexPath.section == min(truncateAfter, eateries.count) && truncateAfter > -1 {
-            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: CarouselMoreEateriesCollectionViewCell.reuse, for: indexPath) as? CarouselMoreEateriesCollectionViewCell else { return UICollectionViewCell() }
-            return cell
+    func setupObserver(_ notificationName: String, completion: @escaping () -> Void) {
+        observer = NotificationCenter.default.addObserver(
+            forName: NSNotification.Name(notificationName),
+            object: nil,
+            queue: .main
+        ) { [weak self] notification in
+            guard self != nil else { return }
+            completion()
         }
-        
-        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: EateryMediumCardCollectionViewCell.reuse, for: indexPath) as? EateryMediumCardCollectionViewCell else { return UICollectionViewCell() }
-        
-        cell.transform = CGAffineTransform(scaleX: 0.1, y: 0.1)
-        
-        UIView.animate(
-            withDuration: 0.6,
-            delay: 0,
-            usingSpringWithDamping: 1,
-            initialSpringVelocity: 2,
-            options: .curveEaseInOut,
-            animations: {
-                cell.transform = .identity
+    }
+
+    // MARK: - Collection View Data Source
+
+    /// Creates and returns the table view data source
+    private func makeDataSource() -> DataSource {
+        let dataSource = DataSource(collectionView: collectionView) { [weak self] (tableview, indexPath, row) in
+            guard let self else { return UICollectionViewCell() }
+
+            switch row {
+            case .eatery(let eatery):
+                guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: EateryMediumCardView.reuse, for: indexPath) as? EateryMediumCardView else { return UICollectionViewCell() }
```

Comment:
I would maybe make this multiple lines, this is a bit hard to read. Ignore this if this is SwiftLint making you do this